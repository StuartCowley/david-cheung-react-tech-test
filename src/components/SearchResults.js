import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ fetchData }) => {
  return (
    <div className="search-results">
      {fetchData &&
        fetchData.length > 0 &&
        fetchData.map((item, index) => (
          <img
            key={index}
            className="image"
            src={item.href}
            alt={item.title}
            title={item.title}
          />
        ))}
    </div>
  );
};

export default SearchResults;

SearchResults.propTypes = {
  fetchData: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string,
      title: PropTypes.string,
    })
  ),
};
