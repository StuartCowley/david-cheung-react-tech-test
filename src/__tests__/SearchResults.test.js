import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = {
    fetchData: [
      { href: "image-href1", title: "image-title1" },
      { href: "image-href2", title: "image-title2" },
      { href: "image-href3", title: "image-title3" },
    ],
  };

  test("Renders as expected", () => {
    const rendered = renderer.create(
      <SearchResults fetchData={validProps.fetchData} />
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert 3 images are present", () => {
    render(<SearchResults fetchData={validProps.fetchData} />);
    const imgs = screen.getAllByRole("img");

    expect(imgs).toHaveLength(3);
  });
});
