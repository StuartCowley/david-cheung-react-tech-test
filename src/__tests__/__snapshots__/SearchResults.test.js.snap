// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Renders as expected 1`] = `
<div
  className="search-results"
>
  <img
    alt="image-title1"
    className="image"
    src="image-href1"
    title="image-title1"
  />
  <img
    alt="image-title2"
    className="image"
    src="image-href2"
    title="image-title2"
  />
  <img
    alt="image-title3"
    className="image"
    src="image-href3"
    title="image-title3"
  />
</div>
`;
